frontend/apps/reader/modules/readerdevicestatus.lua

```diff
@@ -0,0 +1,123 @@
+                            text = _("Dismiss"),
+                            callback = function()
+                                UIManager:close(low_battery_info)
+                                powerd:setDissmisBatteryStatus(true)
+                                UIManager:scheduleIn(300, self.check_low_battery)
+                            end,
+                        },
+                    },
+                }
+            }
+            UIManager:show(low_battery_info)
+            return
+        elseif powerd:getDissmisBatteryStatus() and battery_capacity > threshold then
+            powerd:setDissmisBatteryStatus(false)
+        end
+        UIManager:scheduleIn(300, self.check_low_battery)
+    end
+
+    if self.battery_watchable then
+        self:startBatteryChecker()
+        self.ui.menu:registerToMainMenu(self)
+    end
+end
+
+function ReaderDeviceStatus:addToMainMenu(menu_items)
+    menu_items.battery = {
+        text = _("Low battery alarm"),
+        sub_item_table = {
+            {
+                text = _("Enable"),
+                checked_func = function()
+                    return G_reader_settings:nilOrTrue("battery_alarm")
+                end,
+                callback = function()
+                    G_reader_settings:flipNilOrTrue("battery_alarm")
+                    if G_reader_settings:nilOrTrue("battery_alarm") then
+                        self:startBatteryChecker()
+                    else
+                        self:stopBatteryChecker()
+                        powerd:setDissmisBatteryStatus(false)
+                    end
+                end,
+            },
+            {
+                text = _("Low battery threshold"),
+                enabled_func = function() return G_reader_settings:nilOrTrue("battery_alarm") end,
+                callback = function()
+                    local SpinWidget = require("ui/widget/spinwidget")
+                    local curr_items = G_reader_settings:readSetting("low_battery_threshold") or 20
+                    local battery_spin = SpinWidget:new {
+                        width = Screen:getWidth() * 0.6,
+                        value = curr_items,
+                        value_min = 5,
+                        value_max = 90,
+                        value_hold_step = 10,
+                        ok_text = _("Set threshold"),
+                        title_text = _("Low battery threshold"),
+                        callback = function(battery_spin)
+                            G_reader_settings:saveSetting("low_battery_threshold", battery_spin.value)
+                            powerd:setDissmisBatteryStatus(false)
+                        end
+                    }
+                    UIManager:show(battery_spin)
+                end,
+            },
+        },
+    }
+end
+
+function ReaderDeviceStatus:startBatteryChecker()
+    if G_reader_settings:nilOrTrue("battery_alarm") and self.battery_watchable and self.check_low_battery then
```

Comment:
You could simplify this.
Above, you always define `self.check_low_battery` , so no more need to check for it (in `function ReaderDeviceStatus:stopBatteryChecker()`, `if self.check_low_battery then` is always true

Or, in :init() , define `self.check_low_battery ` only `if self.battery_watchable `, and use `if self.battery_watchable` in the tests around here.
Or forget about `self.battery_watchable` and just use `self.check_low_battery` as a function and a flag.
```
if powerd:getCapacity() > 0 or powerd:isCharging() then
    self.check_low_battery = function()  ... end
else
    self.check_low_battery = nil
end
```